src/gt4py/next/iterator/ir.py

```diff
@@ -40,7 +40,8 @@ def __hash__(self) -> int:
         return hash(type(self)) ^ hash(
             tuple(
                 hash(tuple(v)) if isinstance(v, list) else hash(v)
-                for v in self.iter_children_values()
+                for (k, v) in self.iter_children_items()
+                if k not in ["location", "type"]
             )
         )
```

Comment:
I think this can be refactored to only call `hash()` once...
```suggestion
        return hash(
            (
                type(self),
                *(
                    tuple(v) if isinstance(v, list) else v
                    for (k, v) in self.iter_children_items()
                    if k not in ["location", "type"]
                ),
            )
        )
```